cmd/gardenlet/app/app.go

```diff
@@ -451,6 +465,133 @@ func (g *garden) Start(ctx context.Context) error {
 	return nil
 }
 
+// TODO(aaronfern): Remove this code after v1.91 has been released.
+func (g *garden) createNewDWDResources(ctx context.Context, seedClient client.Client) error {
+	// Fetch all namespaces
+	namespaceList := &corev1.NamespaceList{}
+	if err := seedClient.List(ctx, namespaceList); err != nil {
+		return err
+	}
+
+	// Filter out all namespaces that are in deletion
+	namespacesNotInDeletion := []corev1.Namespace{}
+	for _, namespace := range namespaceList.Items {
+		if namespace.DeletionTimestamp == nil && namespace.Status.Phase != corev1.NamespaceTerminating {
+			namespacesNotInDeletion = append(namespacesNotInDeletion, namespace)
+		}
+	}
+
+	tasks := []flow.TaskFn{}
+	for _, namespace := range namespacesNotInDeletion {
```

Comment:
There is not really a benefit for first filtering the namespaces not in deletion and then iterating over them, is there? You can do everything you need to do in a loop over all namespaces (and `continue` if a namespace is already in deletion).